### ISSUE TITLE
Error with initial compilation of CubeMX-generated Makefile project using FreeRTOS middleware. 

### ISSUE THREAD

**Issue by j0of** [STMicroelectronics/stm32_mw_freertos#5]
Hello, I would like to preface this issue by saying that I have already opened an issue on the ST forum regarding this bug, however it is yet to be solved: https://community.st.com/t5/stm32cubemx-mcus/unable-to-compile-freertos-project/m-p/710692
If this is the wrong place to be reporting this issue, kindly let me know and redirect me to a more suitable source :)

**Describe the set-up**
 * The board (either ST RPN reference or your custom board). --> STM32F103C8T6
 * IDE or at least the compiler and its version. --> STM32CubeMX + GNU gcc/Makefile

**Describe the bug**
--> Makefile projects with FreeRTOS enabled contain errors/do not compile immediately after code generation. Compilation error (make): 
```
In file included from Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:34:
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/freertos_os2.h:31:10: error: #include expects "FILENAME" or <FILENAME>
   31 | #include CMSIS_device_header
      |          ^~~~~~~~~~~~~~~~~~~
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/freertos_os2.h:134:4: error: #error "Definition INCLUDE_xTaskGetCurrentTaskHandle must equal 1 to implement Thread Management API."
  134 |   #error "Definition INCLUDE_xTaskGetCurrentTaskHandle must equal 1 to implement Thread Management API."
      |    ^~~~~
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c: In function 'SVC_Setup':
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:179:3: error: implicit declaration of function 'NVIC_SetPriority' [-Wimplicit-function-declaration]
  179 |   NVIC_SetPriority (SVCall_IRQ_NBR, 0U);
      |   ^~~~~~~~~~~~~~~~
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:80:36: error: 'IRQn_Type' undeclared (first use in this function)
   80 | #define SVCall_IRQ_NBR            (IRQn_Type) -5        /* SVCall_IRQ_NBR added as SV_Call handler name is not the same for CM0 and for all other CMx */
      |                                    ^~~~~~~~~
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:179:21: note: in expansion of macro 'SVCall_IRQ_NBR'
  179 |   NVIC_SetPriority (SVCall_IRQ_NBR, 0U);
      |                     ^~~~~~~~~~~~~~
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:80:36: note: each undeclared identifier is reported only once for each function it appears in
   80 | #define SVCall_IRQ_NBR            (IRQn_Type) -5        /* SVCall_IRQ_NBR added as SV_Call handler name is not the same for CM0 and for all other CMx */
      |                                    ^~~~~~~~~
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:179:21: note: in expansion of macro 'SVCall_IRQ_NBR'
  179 |   NVIC_SetPriority (SVCall_IRQ_NBR, 0U);
      |                     ^~~~~~~~~~~~~~
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c: In function 'OS_Tick_GetCount':
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:401:19: error: 'SysTick' undeclared (first use in this function)
  401 |   uint32_t load = SysTick->LOAD;
      |                   ^~~~~~~
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c: In function 'OS_Tick_GetOverflow':
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:407:12: error: 'SysTick' undeclared (first use in this function)
  407 |   return ((SysTick->CTRL >> 16) & 1U);
      |            ^~~~~~~
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c: In function 'OS_Tick_GetInterval':
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:412:11: error: 'SysTick' undeclared (first use in this function)
  412 |   return (SysTick->LOAD + 1U);
      |           ^~~~~~~
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c: In function 'OS_Tick_GetOverflow':
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:408:1: warning: control reaches end of non-void function [-Wreturn-type]
  408 | }
      | ^
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c: In function 'OS_Tick_GetCount':
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:403:1: warning: control reaches end of non-void function [-Wreturn-type]
  403 | }
      | ^
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c: In function 'OS_Tick_GetInterval':
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:413:1: warning: control reaches end of non-void function [-Wreturn-type]
  413 | }
      | ^
make: *** [Makefile:182: build/cmsis_os2.o] Error 1
```

**How To Reproduce**
1. Indicate the global behavior of your application project.
Compilation error.
2. The modules that you suspect to be the cause of the problem (Driver, BSP, MW ...).
CMSIS/FreeRTOS (Middleware)
3. The use case that generates the problem.
Unsure as to what this means?
4. How we can reproduce the problem.
Generate a Makefile project with FreeRTOS middleware enabled using STM32CubeMX. Open the project in a terminal, then run `make`.

**Additional context**
I was able to generate and compile projects with the same settings in the past without issues.
In regards to the following error message:
```
In file included from Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:34:
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/freertos_os2.h:31:10: error: #include expects "FILENAME" or <FILENAME>
   31 | #include CMSIS_device_header
      |          ^~~~~~~~~~~~~~~~~~~
```
I have attempted to solve this by replacing the line `#include CMSIS_device_header` in file Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/freertos_os2 with the line `#include "stm32f103x6.h"`, which can be found in directory Drivers/CMSIS/Devices/STM32F1xx/Include. This seems to have solved the error, but a few still remained. Namely,
```
Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/freertos_os2.h:134:4: error: #error "Definition INCLUDE_xTaskGetCurrentTaskHandle must equal 1 to implement Thread Management API."
  134 |   #error "Definition INCLUDE_xTaskGetCurrentTaskHandle must equal 1 to implement Thread Management API."
      |    ^~~~~
```
I attempted to solve this error by editing the line `#define INCLUDE_xTaskGetCurrentTaskHandle 0` located in file Middlewares/Third_Party/FreeRTOS/Source/include/FreeRTOS.h to `#define INCLUDE_xTaskGetCurrentTaskHandle 1`. This change resulted in the following compilation error:
```
/usr/lib/gcc/arm-none-eabi/14.1.0/../../../../arm-none-eabi/bin/ld: build/port.o: in function `SysTick_Handler':
/home/joof/projects/ttt/Middlewares/Third_Party/FreeRTOS/Source/portable/GCC/ARM_CM3/port.c:427: multiple definition of `SysTick_Handler'; build/cmsis_os2.o:/home/joof/projects/ttt/Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2/cmsis_os2.c:159: first defined here
/usr/lib/gcc/arm-none-eabi/14.1.0/../../../../arm-none-eabi/bin/ld: warning: build/ttt.elf has a LOAD segment with RWX permissions
collect2: error: ld returned 1 exit status
make: *** [Makefile:190: build/ttt.elf] Error 1
```
**Screenshots**

**Comment by TOUNSTM** [comment]
Hello @j0of,

Thank you for your contribution. You are absolutely right about this point.
Actually, the point you raised has already been dealt and fixed internally. the issue is related to the `STM32CubeMx` generated code. The fix will be made available in the frame of `STM32CubeMX` release `6.13.0`.

Please allow me to close this issue. We look forward to hearing from you again.
With regards,

**Comment by TOUNSTM** [comment]
See also https://github.com/STMicroelectronics/STM32CubeF1/issues/68
